astroid/brain/brain_six.py

```diff
@@ -189,6 +189,29 @@ def transform_six_add_metaclass(node):
             return node
 
 
+def _looks_like_nested_from_six_with_metaclass(node):
+    if len(node.bases) != 1:
+        return False
+    base = node.bases[0]
+    if not isinstance(base, nodes.Call):
+        return False
+    try:
+        func = next(base.func.infer())
```

Comment:
Unfortunately we cannot use inference in predicate functions. Here you would need o unpack the arguments for checking if they refer to `six.with_metaclass` or not (by unpacking I mean either checking explicitly for `Name` or `Attribute` instances that match against `six.with_metaclass`)

Reply:
Should be good now. Let me know if it is still not what like you meant.